SIMA/SIMA/FSM_Groupie.cpp

```diff
@@ -71,22 +84,29 @@ void FSM_groupie::update()
 
 void FSM_groupie::checkObstacle()
 {
-    long distance = ultrasonicSensor.readDistance();
+    long distanceL = leftUltrasonicSensor.readDistance();
+    long distanceR = rightUltrasonicSensor.readDistance();
+    //Serial.print("distL: ");
+    //Serial.println(distanceL);
+    //Serial.print("distR: ");
+    //Serial.println(distanceR);
+
     // Checks if obstacle is closer than 10 cm
-    if (distance < obstacleDistance)
+    if (distanceL < obstacleDistance || distanceR < obstacleDistance)
     {
         // If a new obstacle is detected, start counting the time
         if (previousState != AVOID_OBSTACLE)
         {
             obstacleStartTime = currentTime;
         }
         previousState = currentState;
+
         currentState = AVOID_OBSTACLE;
     }
     else
     {
         // If the robot finished avoiding the obstacle, add the time to the total obstacle time
-        if (previousState == AVOID_OBSTACLE)
+        if (previousState == AVOID_OBSTACLE);
         {
             totalObstacleTime += currentTime - obstacleStartTime;
         }
```

Comment:
There seems to be an unexpected semicolon in `FSM_groupie::checkObstacle()`:
```suggestion
        if (previousState == AVOID_OBSTACLE)
        {
            totalObstacleTime += currentTime - obstacleStartTime;
        }
```